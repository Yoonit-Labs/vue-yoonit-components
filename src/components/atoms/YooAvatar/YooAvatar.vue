<template lang="pug">
  div(
    :class="['yoo-avatar', ...takeModifier, ...takeFill]"
  )
    p(
      :class="['yoo-avatar__text', ...takeTextColor]"
      v-if="takeInitials && !takeImage"
    ) {{ takeInitials }}

    img(
      v-else
      :src="takeImage"
      :class="['yoo-avatar', ...takeModifier]"
    )
</template>

<script>
import { StringParser } from '../../bosons'

export default {
  name: 'YooAvatar',
  props: {
    size: {
      type: String,
      default: 'medium',
      validator: value =>
        [
          'small',
          'medium',
          'large',
          'xlarge'
        ]
          .includes(value)
    },
    fill: {
      type: String,
      default: 'neutral',
      validator: value =>
        [
          'primary',
          'danger',
          'light',
          'dark',
          'neutral'
        ]
          .includes(value)
    },
    text: {
      type: String,
      required: false
    },
    textColor: {
      type: String,
      default: 'light',
      validator: value =>
        [
          'light',
          'dark'
        ]
          .includes(value)
    },
    image: {
      type: String,
      required: false
    }
  },
  data: () => ({
  }),
  created () {},
  computed: {
    takeModifier () {
      const block = 'yoo-avatar'
      const classList = []
      switch (this.size) {
        case 'small':
          classList
            .push(`${block}--${this.size}`)
          break
        case 'medium':
          classList
            .push(`${block}--${this.size}`)
          break
        case 'large':
          classList
            .push(`${block}--${this.size}`)
          break
        case 'xlarge':
          classList
            .push(`${block}--${this.size}`)
          break
      }
      return classList
    },
    takeFill () {
      return [`yoo-avatar--${this.fill}`]
    },
    takeInitials () {
      return StringParser.generateNameInitials(this.text)
    },
    takeTextColor () {
      return [`yoo-avatar__text--${this.textColor}`]
    },
    takeImage () {
      return this.image
    }
  },
  mounted () {},
  updated () {},
  beforeDestroy () {},
  destroyed () {},
  components: {},
  methods: {},
  filters: {},
  watch: {}
}
</script>

<style src="./YooAvatar.sass" lang="sass" scoped></style>
