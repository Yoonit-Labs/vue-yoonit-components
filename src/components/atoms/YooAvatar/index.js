import YooAvatar from './YooAvatar.vue'
export default YooAvatar
