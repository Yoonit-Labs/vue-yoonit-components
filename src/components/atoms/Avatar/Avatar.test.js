/**
 * ██╗   ██╗ ██████╗  ██████╗ ███╗   ██╗██╗████████╗
 * ╚██╗ ██╔╝██╔═══██╗██╔═══██╗████╗  ██║██║╚══██╔══╝
 *  ╚████╔╝ ██║   ██║██║   ██║██╔██╗ ██║██║   ██║
 *   ╚██╔╝  ██║   ██║██║   ██║██║╚██╗██║██║   ██║
 *    ██║   ╚██████╔╝╚██████╔╝██║ ╚████║██║   ██║
 *    ╚═╝    ╚═════╝  ╚═════╝ ╚═╝  ╚═══╝╚═╝   ╚═╝
 *
 * https://yoonit.dev - about@yoonit.dev
 *
 * Vue Yoonit Components
 * VueJS Atomic Design System framework
 *
 * Vitória Costa, Tiago Brito, Fernando Junior, Sabrina Sampaio, Gabriel Mule, Gabriel Moraes, Gabriel Rizzo & Luigui Delyer @ 2020-2021
 */

import { shallowMount } from '@vue/test-utils'
import YooAvatar from '@/components/atoms/Avatar/Avatar.vue'
import PropsConfig from '@/components/atoms/Avatar/Avatar.config'

const classBlock = 'yoo-avatar'
const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(YooAvatar, {
    slots: { default: SlotText }
  })
}

describe('YooAvatar Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    expect(wrapper.find('.yoo-avatar').exists()).toBe(true)
  })

  describe('Props', () => {
    describe('size', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.size.options.includes(YooAvatar.props.size.default)).toBe(true)
      })
      PropsConfig.size.options.forEach(size => {
        it(`Includes variation class: .${classBlock}--${size}`, async () => {
          await wrapper.setProps({ size })
          expect(wrapper.find(`.${classBlock}--${size}`).exists()).toBe(true)
        })
      })
    })
    // describe Props
    describe('fill', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.fill.options.includes(YooAvatar.props.fill.default)).toBe(true)
      })
      PropsConfig.fill.options.forEach(fill => {
        it(`Includes variation class: .${classBlock}--${fill}`, async () => {
          await wrapper.setProps({ fill })
          expect(wrapper.find(`.${classBlock}--${fill}`).exists()).toBe(true)
        })
      })
    })

    describe('textColor', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.textColor.options.includes(YooAvatar.props.textColor.default)).toBe(true)
      })
      PropsConfig.textColor.options.forEach(textColor => {
        it(`Includes variation class: .${classBlock}__text--${textColor}`, async () => {
          await wrapper.setProps({ text: 'text-value', textColor })
          expect(wrapper.find(`.${classBlock}__text--${textColor}`).exists()).toBe(true)
        })
      })
    })
  })
})
