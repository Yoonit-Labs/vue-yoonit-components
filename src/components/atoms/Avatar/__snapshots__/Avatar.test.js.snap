// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooAvatar Component Matches Snapshot 1`] = `<div class="yoo-avatar yoo-avatar--medium yoo-avatar--neutral"><img class="yoo-avatar yoo-avatar--medium"></div>`;
